esb/plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/debugger/debugpoint/builder/impl/AbstractESBDebugPointBuilder.java

```diff
@@ -0,0 +1,902 @@
+        } else if (mediator instanceof SwitchMediator) {
+            return ((SwitchMediator) mediator).getInputConnector();
+        } else if (mediator instanceof ThrottleMediator) {
+            return ((ThrottleMediator) mediator).getInputConnector();
+        } else if (mediator instanceof XQueryMediator) {
+            return ((XQueryMediator) mediator).getInputConnector();
+        } else if (mediator instanceof XSLTMediator) {
+            return ((XSLTMediator) mediator).getInputConnector();
+        } else if (mediator instanceof FastXSLTMediator) {
+            return ((FastXSLTMediator) mediator).getInputConnector();
+        } else if (mediator instanceof BAMMediator) {
+            return ((BAMMediator) mediator).getInputConnector();
+        } else if (mediator instanceof Sequence) {
+            return ((Sequence) mediator).getInputConnector();
+        } else if (mediator instanceof CallMediator) {
+            return ((CallMediator) mediator).getInputConnector();
+        } else if (mediator instanceof LoopBackMediator) {
+            return ((LoopBackMediator) mediator).getInputConnector();
+        } else if (mediator instanceof RespondMediator) {
+            return ((RespondMediator) mediator).getInputConnector();
+        } else if (mediator instanceof ConditionalRouterMediator) {
+            return ((ConditionalRouterMediator) mediator).getInputConnector();
+        } else if (mediator instanceof ValidateMediator) {
+            return ((ValidateMediator) mediator).getInputConnector();
+        } else if (mediator instanceof BeanMediator) {
+            return ((BeanMediator) mediator).getInputConnector();
+        } else if (mediator instanceof EJBMediator) {
+            return ((EJBMediator) mediator).getInputConnector();
+        } else if (mediator instanceof URLRewriteMediator) {
+            return ((URLRewriteMediator) mediator).getInputConnector();
+        } else if (mediator instanceof TransactionMediator) {
+            return ((TransactionMediator) mediator).getInputConnector();
+        } else if (mediator instanceof ForEachMediator) {
+            return ((ForEachMediator) mediator).getInputConnector();
+        } else if (mediator instanceof BuilderMediator) {
+            return ((BuilderMediator) mediator).getInputConnector();
+        } else if (mediator instanceof PublishEventMediator) {
+            return ((PublishEventMediator) mediator).getInputConnector();
+        } else if (mediator instanceof CloudConnectorOperationImpl) {
+            return ((CloudConnectorOperationImpl) mediator).getInputConnector();
+        }
+        throw new IllegalArgumentException("Invalid/Unknown Mediator type found : " + mediator.toString());
+    }
+
+    /**
+     * This method returns the OutputConnector of the given mediator.
+     * 
+     * @param mediator
+     * @return
+     */
+    public static OutputConnector getOutputConnector(Mediator mediator) {
```

Comment:
Can we add this getOutputConnector(Mediator) method to a common util class so that we can reuse this method